pkg/virt-handler/device-manager/device_controller.go

```diff
@@ -34,51 +34,120 @@ import (
 	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 )
 
-var permanentDevicePluginPaths = map[string]string{
-	"kvm":       "/dev/kvm",
-	"tun":       "/dev/net/tun",
-	"vhost-net": "/dev/vhost-net",
-	"sev":       "/dev/sev",
+var defaultBackoffTime = []time.Duration{1 * time.Second, 2 * time.Second, 5 * time.Second, 10 * time.Second}
+
+type controlledDevice struct {
+	devicePlugin GenericDevice
+	started      bool
+	stopChan     chan struct{}
+	backoff      []time.Duration
 }
 
-type DeviceControllerInterface interface {
-	Initialized() bool
-	RefreshMediatedDevicesTypes()
+func (c *controlledDevice) Start() {
+	if c.started {
+		return
+	}
+
+	stop := make(chan struct{})
+
+	logger := log.DefaultLogger()
+	dev := c.devicePlugin
+	deviceName := dev.GetDeviceName()
+	logger.Infof("Starting a device plugin for device: %s", deviceName)
+	retries := 0
+
+	backoff := c.backoff
+	if backoff == nil {
+		backoff = defaultBackoffTime
+	}
+
+	go func() {
+		for {
+			err := dev.Start(stop)
+			if err != nil {
+				logger.Reason(err).Errorf("Error starting %s device plugin", deviceName)
+				retries = int(math.Min(float64(retries+1), float64(len(backoff)-1)))
+			} else {
+				retries = 0
+			}
+
+			select {
+			case <-stop:
+				// Ok we don't want to re-register
+				return
+			case <-time.After(backoff[retries]):
+				// Wait a little and re-register
+				continue
+			}
+		}
+	}()
+
+	c.stopChan = stop
+	c.started = true
 }
 
-type DeviceController struct {
-	devicePlugins      map[string]ControlledDevice
-	devicePluginsMutex sync.Mutex
-	host               string
-	maxDevices         int
-	backoff            []time.Duration
-	virtConfig         *virtconfig.ClusterConfig
-	stop               chan struct{}
-	mdevTypesManager   *MDEVTypesManager
-	clientset          k8scli.CoreV1Interface
+func (c *controlledDevice) Stop() {
+	if !c.started {
+		return
+	}
+	close(c.stopChan)
```

Comment:
Shouldn't you send a signal via the channel so the controlled device performs a graceful shutdown ?